common/src/main/java/dev/ftb/mods/ftblibrary/sidebar/SidebarButtonManager.java

```diff
@@ -1,187 +1,169 @@
 package dev.ftb.mods.ftblibrary.sidebar;
 
-import com.google.gson.*;
-import com.google.gson.stream.JsonWriter;
-import dev.architectury.platform.Platform;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonNull;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonParser;
+import com.mojang.logging.LogUtils;
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.DataResult;
+import com.mojang.serialization.JsonOps;
 import dev.ftb.mods.ftblibrary.FTBLibrary;
+import dev.ftb.mods.ftblibrary.api.sidebar.SidebarButtonCreatedEvent;
+import dev.ftb.mods.ftblibrary.config.FTBLibraryClientConfig;
+import dev.ftb.mods.ftblibrary.snbt.config.StringSidebarMapValue;
+import dev.ftb.mods.ftblibrary.util.MapUtils;
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.server.packs.resources.Resource;
 import net.minecraft.server.packs.resources.ResourceManager;
 import net.minecraft.server.packs.resources.ResourceManagerReloadListener;
+import org.slf4j.Logger;
 
-import java.io.File;
-import java.io.FileReader;
-import java.io.FileWriter;
+import java.io.BufferedReader;
 import java.io.IOException;
 import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Comparator;
 import java.util.HashMap;
+import java.util.LinkedList;
 import java.util.List;
 import java.util.Map;
+import java.util.function.BiConsumer;
+import java.util.stream.Collectors;
 
 
 public enum SidebarButtonManager implements ResourceManagerReloadListener {
```

Comment:
Although this is OK, a class extending `SimpleJsonResourceReloadListener` is probably better here - you get the JSON loading for free and just have to override one `apply()` method.  See e.g. https://github.com/FTBTeam/FTB-Team-Bases/blob/main/common/src/main/java/dev/ftb/mods/ftbteambases/data/definition/BaseDefinitionManager.java#L50